extensions/wikia/Email/Controller/FounderController.class.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\Check;
+use Email\EmailController;
+use Email\Fatal;
+
+abstract class FounderController extends EmailController {
+
+	/** @var \Title */
+	protected $pageTitle;
+
+	protected $previousRevId;
+	protected $currentRevId;
+
+	public function initEmail() {
+		// This title is for the article being commented upon
+		$titleText = $this->request->getVal( 'pageTitle' );
+		$titleNamespace = $this->request->getVal( 'pageNs' );
+
+		$this->pageTitle = \Title::newFromText( $titleText, $titleNamespace );
+
+		$this->previousRevId = $this->request->getVal( 'previousRevId' );
+		$this->currentRevId = $this->request->getVal( 'currentRevId' );
+
+		$this->assertValidParams();
+	}
+
+	/**
+	 * Validate the params passed in by the client
+	 */
+	private function assertValidParams() {
+		$this->assertValidTitle();
+		$this->assertValidRevisionIds();
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidTitle() {
+		if ( !$this->pageTitle instanceof \Title ) {
+			throw new Check( "Invalid value passed for title" );
+		}
+
+		if ( !$this->pageTitle->exists() ) {
+			throw new Check( "Title doesn't exist." );
+		}
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidRevisionIds() {
+		if ( empty( $this->previousRevId ) ) {
+			throw new Check( "Invalid value passed for previousRevId" );
+		}
+
+		if ( empty( $this->currentRevId ) ) {
+			throw new Check( "Invalid value passed for currentRevId" );
+		}
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->getCurrentUserName(),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'summary' => $this->getSummary(),
+			'buttonText' => $this->getChangesLabel(),
+			'buttonLink' => $this->getChangesLink(),
+			'contentFooterMessages' => [
+				$this->getFooterEncouragement(),
+				$this->getFooterArticleLink(),
+				$this->getFooterAllChangesLink(),
+			],
+			'details' => $this->getDetails(),
+			'hasContentFooterMessages' => true
```

Comment:
I've looked at it several times because I don't like it either.  However its not possible in mustache AFAICT.  The code is this:
```mustache
{{# hasContentFooterMessages}}
	<tr>
		<td colspan="2">
			<div class="content-footer-messages">
				{{#contentFooterMessages}}
					<p>{{{.}}}</p>
				{{/contentFooterMessages}}
			</div>
		</td>
	</tr>
{{/ hasContentFooterMessages}}
```
We only want the ```<tr>``` stuff if we actually have an array of values in ```contentFooterMessages```.  However we only want it once, and mustache only gives you one behavior from ```{{#foo}}...{{/foo}}```; a simple conditional if ```foo``` is a scalar value or a loop if ```foo``` is an array.